lib/transition/unpack.c

```diff
@@ -164,6 +185,249 @@ unpack_synapse(crm_graph_t * new_graph, xmlNode * xml_synapse)
     return new_synapse;
 }
 
+static gboolean
+has_injected_metadata_for(crm_action_t *action, GHashTable *injected_metadata)
+{
+    const char *node = NULL;
+    char *key = NULL;
+    gboolean result = FALSE;
+
+    CRM_CHECK(action, return FALSE);
+
+    if (!action->rsc_info) {
+        return FALSE;
+    }
+
+    node = crm_meta_value(action->params, XML_LRM_ATTR_TARGET);
+
+    key = crm_generate_key(4, action->rsc_info->class, action->rsc_info->provider,
+                              action->rsc_info->type, node);
```

Comment:
I also thought it would be handy to have pointer to a graph in synapse instead of passing it to the various functions. To do so I changed the crm_graph_t declaration. I hope backward compatibility is still saved.